myTinyURL/convert/convert.py

```diff
@@ -31,12 +33,19 @@ def long2short(cls, http_url):
             The response(SITE_URL + short_id) we just create
             type: str
         """
-        lock = Lock()
-        lock.acquire()
-        short_id = Convert.gen_short_id()
-        lock.release()
-        record = URL(http_url=http_url, short_id=short_id)
-        record.save()
+        with transaction.atomic():
+            latest = LatestId.objects.select_for_update().get(id=1)
+            latest_short_id = latest.short_id
+            short_id = str(int(latest_short_id) + 1).rjust(Convert.SHORT_ID_LENGTH, '0')
```

Comment:
I'd suggest to use type `int` for short_id in the database. It's the source of truth about what's the latest id. 
How to convert it to a string and convert a string back to int should be handled in application level, a.k.a. Convert() function.

Current the approach is error prone because you may forget what's the format of how you store the short_id. Anyone else who works on this codebase will also confuse about it. 

Using int in db will make your program and model elegant and robust.